### ISSUE TITLE
[DataObjects] Relation fields inside Fieldcollections are not saved on update 

### ISSUE THREAD

**Issue by dvesh3** [pimcore/pimcore#6733]
### Expected behavior
Relation fields inside Fieldcollections should be saved on update
### Actual behavior
Relation fields inside Fieldcollections are not saved on update , **when fieldcollection setter is not called** 
### Steps to reproduce
```php
    $man = Manufacturer::getById(80);
    $fc = $man->getTestfc();

    $item0 = $fc->get(0);
    $item0->setInputtest("test 000"); // working and saved
    $item0->setManytoonetest(News::getById(766)); // updated in dump but not saved
    $item0->setManytomanytest([News::getById(766)]);  // updated in dump but not saved
    dump($item0);

    $item1 = $fc->get(1);
    $item1->setInputtest("test 111"); // working and saved
    $item1->setManytoonetest(News::getById(766));  // updated in dump but not saved
    $item1->setManytomanytest([News::getById(766)]);  // updated in dump but not saved
    dump($item1);

    $man->save();
```
https://talk.pimcore.org/t/saving-field-collections/3680/3

**Comment by dvesh3** [comment]
Also the case with Localized Relations

**Comment by dvesh3** [comment]
closed by #6734
